doc/introduction/measurements.rst

```diff
@@ -354,6 +354,53 @@ The transform can be applied to a QNode as follows:
     If the ``defer_measurements`` transform is used in analytic mode, ``backprop`` is also a viable
     option.
 
+.. _tree_traversal:
+
+The tree-traversal algorithm
+****************************
+
+Dynamic circuit execution is akin to traversing a binary tree where each MCM
+corresponds to a node and groups of gates between the MCMs correspond to edges.
+The :func:`~.pennylane.dynamic_one_shot` approach picks a branch of the tree randomly
+and simulates it from beginning to end.
+This is wasteful in many cases; the same branch is simulated many times
+when there are more shots than branches for example.
+The tree-traversal algorithm does away with such redundancy while retaining the
+exponential gains in memory of the one-shot approach compared with the deferred
+measurement principle, among other advantages.
+
+Briefly, it proceeds from the top of the tree and begins splitting the circuit
+into two: a circuit running to the next MCM replacing the MCM by a sample
+measurement, and a circuit consisting of the rest of the circuit. A key point is that
+all samples are collected at once, removing the need to run the circuit several times.
+The samples are converted to counts and a copy of the state vector is made at each node.
+We loop over the possible branches, collapsing the state vector and pursuing the calculation
+until the next MCM. Since many counts come out to be zero in practice, it is possible
```

Comment:
```suggestion
into two: a circuit running to the next mid-circuit measurement, replacing the mid-circuit measurement by a sample
measurement, and a circuit consisting of the rest of the circuit. A key point is that
all samples are collected at once, removing the need to run the circuit several times.
The samples are converted to counts and a copy of the state vector is made at each node.
We loop over the possible branches, collapsing the state vector and pursuing the calculation
until the next mid-circuit measurement. Since many counts come out to be zero in practice, it is possible
```